MovieApp/MovieCover.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  MovieDetailsWithCoverView.swift
+//  MovieApp
+//
+//  Created by endava-bootcamp on 02.04.2023..
+//
+
+import Foundation
+import PureLayout
+import MovieAppData
+
+class MovieCover: UIView {
+    
+    private var poster: UIImageView!
+    private var userRatingNumber: UILabel!
+    private var userRating: UILabel!
+    private var year: UILabel!
+    private var date: UILabel!
+    private var category: MovieCategoryModel!
+    private var genre: UILabel!
+    private var movieDetails: MovieDetailsModel!
+    private var movieTitle: UILabel!
+    private var favourite: UIButton!
+    private var duration: UILabel!
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    init() {
+        super.init(frame: CGRect())
+        movieDetails = MovieUseCase().getDetails(id: 111161)
```

Comment:
You should fetch movie details in the ViewController and forward data to custom views. UIView classes should only contain view related logic.